theano/gof/op.py

```diff
@@ -821,6 +821,14 @@ def __eq__(self, other):
         else:
             return NotImplemented
 
+    def prepare_node(self, node):
+        """
+        Make any special modifications that the Op needs before doing
+        make_thunk().
```

Comment:
Tell that is if it can modify in place this node if this don't change its semantic. Like adding tag and precomputed stuff.

But tell that it can return a new node to use it this is needed.